src/OrchardCore/OrchardCore.Deployment.Core/Services/DeploymentManager.cs

```diff
@@ -23,9 +23,11 @@ public class DeploymentManager : IDeploymentManager
 
         public async Task ExecuteDeploymentPlanAsync(DeploymentPlan deploymentPlan, DeploymentPlanResult result)
         {
-            foreach (var step in deploymentPlan.DeploymentSteps)
+            var orderedDeploymentSources = new DeploymentSourceSortedList(_deploymentSources);
+
+            foreach (var source in orderedDeploymentSources)
```

Comment:
So here, by using an `IComparer<IDeploymentSource>` as suggested above, here we would just need to use an `_deploymentSources.OrderBy(DeploymentSourceComparer)`

Reply:
Aha, I understand why you want the above checks ;)